odoo/tools/pdf.py

```diff
@@ -32,12 +44,78 @@ def rotate_pdf(pdf):
     :param pdf: a PDF to rotate
     :return: a PDF rotated
     '''
-    writer = PyPDF2.PdfFileWriter()
-    reader = PyPDF2.PdfFileReader(io.BytesIO(pdf), strict=False)
+    writer = PdfFileWriter()
+    reader = PdfFileReader(io.BytesIO(pdf), strict=False)
     for page in range(0, reader.getNumPages()):
         page = reader.getPage(page)
         page.rotateClockwise(90)
         writer.addPage(page)
     with io.BytesIO() as _buffer:
         writer.write(_buffer)
         return _buffer.getvalue()
+
+
+class OdooPdfFileReader(PdfFileReader):
+    # OVERRIDE of PdfFileReader to add the management of multiple embedded files.
+
+    def getAttachment(self, name):
+        for i in range(0, len(self.trailer["/Root"]["/Names"]["/EmbeddedFiles"]["/Names"]), 2):
+            if self.trailer["/Root"]["/Names"]["/EmbeddedFiles"]["/Names"][i+1].getObject()["/F"]==name:
+                return self.trailer["/Root"]["/Names"]["/EmbeddedFiles"]["/Names"][i+1].getObject()["/EF"]["/F"].getObject().getData()
```

Comment:
`yield`?

Reply:
We return the first attachment with the name 'name'. There shouldn't be multiple attachment with the same name.

I added documented the function to be sure.

Reply:
why not  `getAttachments()` that will return a dictionary <filename> -> <filecontent> instead of two methods

Reply:
More flexibility ? Since pdf.py is in tools it allows to list all files and only extract the one needed.